hardhat.config.ts

```diff
@@ -0,0 +1,199 @@
+import { HardhatUserConfig, task } from "hardhat/config";
+import "@nomicfoundation/hardhat-toolbox-viem";
+import "@nomicfoundation/hardhat-foundry";
+import "hardhat-deploy";
+import "dotenv/config";
+import "./tasks";
+import {
+  ERC4626InstanceDeployConfigInput,
+  FactoryDeployConfigInput,
+  StETHInstanceDeployConfigInput,
+} from "./tasks";
+import { RETHInstanceDeployConfigInput } from "./tasks/deploy/instances/reth";
+
+const TEST_FACTORY: FactoryDeployConfigInput = {
```

Comment:
I might be missing something, but why are the constants in this file prefixed with `TEST_`. Do you anticipate this changing when we go to mainnet? IMO, the factory isn't a test contract, but `MockLido` and `MockERC4626` are.

Reply:
We should absolutely go through these carefully before mainnet, IIRC I copied the bulk of these parameters from our prior testing deploys hence the name. 

I've set up the hardhat network to have its own `deployments.local.json` file, and was thinking we would have dedicated factory/pool configurations for it as well. No strong preferences either way.